script/hassfest/dependencies.py

```diff
@@ -156,15 +157,16 @@ def visit_Attribute(self, node):
 
 def calc_allowed_references(integration: Integration) -> set[str]:
     """Return a set of allowed references."""
+    manifest = integration.manifest
     allowed_references = (
         ALLOWED_USED_COMPONENTS
-        | set(integration.manifest.get("dependencies", []))
-        | set(integration.manifest.get("after_dependencies", []))
+        | set(manifest.get("dependencies", []))
+        | set(manifest.get("after_dependencies", []))
     )
 
     # Discovery requirements are ok if referenced in manifest
     for check_domain, to_check in DISCOVERY_INTEGRATIONS.items():
-        if any(check in integration.manifest for check in to_check):
+        if any(check in manifest for check in to_check):
```

Comment:
Because integration.manifest is now typed better, this change is technically not required anymore.
I'm not sure if it adds anything. It also doesn't hurt so...